### PR TITLE
1 make protocol handler portable

### LINKED ISSUE #1: Make protocol handler portable

### PR DESCRIPTION
**Pull request by Keaton-Clark**
With docker installed and in the docker group, run `make debian` and you will be dropped in a debian instance where you can `import dddb` and use `dddb.video.{encode, decode}`. Install on a debian system with `dpkg -i python3-dddb.deb`. If there are any missing dependencies let me know and I will add them to the debian package. If we want auto completion and type hinting in lsps for our module I can add that as well.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM debian:latest
+RUN apt update && apt install -y curl libopencv-dev build-essential ffmpeg clang libssl-dev libclang-dev
```
Comment (lgactna) [inline Dockerfile]:
```suggestion
RUN DEBIAN_FRONTEND=noninteractive apt update && apt install -yq --no-install-recommends curl libopencv-dev build-essential ffmpeg clang libssl-dev libclang-dev ca-certificates python3
```

`ca-certificates` is needed else curl errors out with `curl: (77) error setting certificate file: /etc/ssl/certs/ca-certificates.crt`. 

`python3` is needed for obvious reasons (+ rustup just won't let the install finish)

apt would also randomly fail to fetch packages prior to setting `DEBIAN_FRONTEND=noninteractive`, not sure exactly why but it works now -- don't know if this was a one-off thing, somebody else wld need to test

i don't care if `-q` flag is set

File: dddb/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+use pyo3::prelude::*;
+use isg_4real::*;
+
+#[pymodule]
+fn dddb(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    let submodule = PyModule::new(_py, "video")?;
+    submodule.add_function(wrap_pyfunction!(encode, m)?)?;
```
Comment (lgactna) [inline dddb/src/lib.rs:7]: obviously i've never worked with pyo3 before but why's the interface

```py
dddb.video.decode({'in_path': 'test.mp4', 'out_path': 'test-new.txt'})
```

and not
```py
dddb.video.decode(in_path = 'test.mp4', out_path = 'test.txt')
```

not a big concern but why does it do this lol

**Review by lgactna — CHANGES_REQUESTED** [review]
congrats, looks like it works lol

image

File: dddb/src/lib.rs (hunk 2 above)
Comment (Keaton-Clark) [reply dddb/src/lib.rs:7]: allows us to pass in an arbitrary argument set

**Review by lgactna — APPROVED** [review]
I love ou ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ 

Good code !

**Pull request merged**